src/main/java/com/google/sps/data/FakeAvailabilityDao.java

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+/** Mimics accessing Datastore to support managing Availability entities. */
+public class FakeAvailabilityDao implements AvailabilityDao {
+  // @param datastore the fake database. Made public because it will only be used
+  // while testing.
+  public LinkedHashMap<Long, Availability> datastore;
```

Comment:
Naming this `datastore` is confusing, even though it is simulating the behavior of datastore. `storedObjects` or just `data` might work better.